### ISSUE TITLE
fail to run uploadAggregateToAppEngine.app

### ISSUE THREAD

**Issue by danbjoseph** [opendatakit/aggregate#317]
macOS 10.13.6 and Java Version 10.0.2 (build 10.0.2+13) x86_64

trying to run `uploadAggregateToAppEngine.app` it appears in finder bar then disappears

thought it might be this:
> Depending on your Java version, security settings may prevent the upload tool from running. A possible workaround is to add `file://` to the Exception Site List.

so tried this:
<img width="894" alt="screen shot 2018-09-19 at 10 35 54 am" src="https://user-images.githubusercontent.com/4806884/45760498-34fbc980-bbf8-11e8-8814-9393954baba1.png">

then tried from terminal and got this:
```
➜  ODK Aggregate sh uploadAggregateToAppEngine.sh
CodeSource Location: /Users/danbjoseph/Desktop/aggregate-testing/ODK Aggregate/ODKAggregateAppEngineUpdater.jar/ODKAggregateAppEngineUpdater.jar
user.dir path: /Users/danbjoseph/Desktop/aggregate-testing/ODK Aggregate/ODKAggregateAppEngineUpdater.jar
Directory of this runnable Jar: /Users/danbjoseph/Desktop/aggregate-testing/ODK Aggregate
Exception in thread "AWT-EventQueue-0" java.lang.ExceptionInInitializerError
	at org.opendatakit.appengine.updater.AppCfgWrapper.locateTokenFile(AppCfgWrapper.java:50)
	at org.opendatakit.appengine.updater.UpdaterWindow.perhapsHasToken(UpdaterWindow.java:504)
	at org.opendatakit.appengine.updater.UpdaterWindow.<init>(UpdaterWindow.java:434)
	at org.opendatakit.appengine.updater.UpdaterWindow$1.run(UpdaterWindow.java:269)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue.access$600(EventQueue.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3107)
	at java.base/java.lang.String.substring(String.java:1873)
	at org.apache.commons.lang.SystemUtils.getJavaVersionAsFloat(SystemUtils.java:1122)
	at org.apache.commons.lang.SystemUtils.<clinit>(SystemUtils.java:818)
	... 18 more
```

**Comment by yanokwa** [comment]
@danbjoseph If you run `java -version` on the command line, what do you get? I'm betting you get Java 11.
```
$ java -version
java version "11" 2018-09-25
Java(TM) SE Runtime Environment 18.9 (build 11+28)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11+28, mixed mode)
```

@ggalmazor, I haven't looked deeply, but my guess is that our code expects versions have periods in them.

**Comment by ggalmazor** [comment]
@yanokwa, @mitchellsundt do you know how was the macOS app developed? Do we have access to the source code?

**Comment by ggalmazor** [comment]
We're using an obsolete Apache Commons Lang to get the host's operating system. I'll replace that with a more modern approach we do in Briefcase for the next release.

**Comment by ggalmazor** [comment]
A PR with the fix has been submitted here: https://github.com/opendatakit/aggregate-components/pull/4

I've created #320 to track updating the updater JAR we will ship in the next Aggregate release.

**Comment by ggalmazor** [comment]
So, PR #321 should deal with the new GAE Updater. The only thing missing would be to study the macOS app, just in case :)

**Comment by yanokwa** [comment]
Closed by #321

**Comment by yanokwa** [comment]
I should add more information, huh? 

@danbjoseph we've fixed this in master and it'll be fixed in the next release of Aggregate. If you need to install Aggregate today, download [ODKAggregateAppEngineUpdater.jar.zip](https://github.com/opendatakit/aggregate/files/2447203/ODKAggregateAppEngineUpdater.jar.zip), unzip it and replace the one you have in your `ODK Aggregate` folder. 

Also, note that you can run Aggregate's installer on any computer the one you are on is giving your trouble.
